### PR TITLE
Adding Android robot teleoperation demo.

### PR DESCRIPTION
**Pull request by gabrik**
- [x] Make the app work
- [x] Remove all the old examples
- [x] Improve configuration

This has been tested with the TurtleBot4, and as already been used during #ZUM23.

### CODE REVIEW

**Comment by gabrik** [comment]
@DariusIMP can you have a look and see if it is Android enough?

File: zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ros/Battery.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        inputStream.readUByte(),
+        inputStream.readUByte(),
+        inputStream.readUByte(),
+        inputStream.readBoolean(),
+        inputStream.readFloatArray(),
+        inputStream.readFloatArray(),
+        inputStream.readString(),
+        inputStream.readString(),
+    ) {
+        //Log.v("BatteryConstructor", "streamPos: ${inputStream.buffer.position()} - ${this.toString()}")
```
Comment (DariusIMP) [inline zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ros/Battery.kt:44]: Can we remove these commented logs? or uncomment them?

File: zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ros/Header.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import android.util.Log
+import com.example.zenohapp.cdr.CDRInputStream
+import com.example.zenohapp.cdr.CDROutputStream
+import com.example.zenohapp.ros.Time
+
+class Header constructor(var time: Time, var frameId: String) {
+
+    constructor(inputStream : CDRInputStream) : this( Time(inputStream), inputStream.readString()) {
+        //Log.v("Header","streamPos: ${inputStream.buffer.position()} - ${this.toString()}")
```
Comment (DariusIMP) [inline zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ros/Header.kt:11]: Same here

File: zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ros/Image.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        stream.readUByte(),
+        stream.readUInt(),
+        stream.readByteArray(),
+    )
+
+    fun toBitmap() : Bitmap {
+        val nrOfPixels = data.size / 3; // Three bytes per pixel.
+        var pixels = IntArray(nrOfPixels){0}
+        for(i in 0..< nrOfPixels) {
+//            val b = data[3*i].toInt()
```
Comment (DariusIMP) [inline zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ros/Image.kt:32]: Can we remove these?

File: zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ui/examples/TeleopFragment.kt [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+    private lateinit var mCameraSubJob : Job
+    private lateinit var mCommandPubJob : Job
+
+    private val actionDockSuffix = "/dock/_action/send_goal"
+    private val cmdVelSuffix = "/cmd_vel"
+    private val batteryStateSuffix = "/battery_state"
+    private val imageRawSuffix = "/oakd/rgb/preview/image_raw"
+    private var angularScale = 0.5
+    private var linearScale = 0.5
+    override fun onTouch(v: View?, event: MotionEvent?): Boolean {
```
Comment (DariusIMP) [inline zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ui/examples/TeleopFragment.kt:90]: So whenever you touch the fragment layout, this callback is going to be triggered and it's going to check if the touch happened on a button?
I'd instead set for each button an implementation for [Button::onTouchEvent(MotionEvent event)](https://developer.android.com/reference/android/widget/TextView#onTouchEvent(android.view.MotionEvent)).

File: zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ui/examples/TeleopFragment.kt [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+        Toast.makeText(
+            activity,
+            errorMsg,
+            Toast.LENGTH_SHORT
+        ).show()
+    }
+
+    override fun onClick(v: View?) {
+        when (v?.id) {
+            R.id.buttonDeclare -> {
```
Comment (DariusIMP) [inline zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ui/examples/TeleopFragment.kt:281]: Here it'd be more efficient to do something like

```kotlin
mDeclareButton.setOnClickListener(v -> {
    declarePublisher()
    startPubSub()
})
```
and same thing for the other buttons

File: zenoh-android/RobotTeleop/zenohapp_menu.jpeg
Comment (DariusIMP) [inline zenoh-android/RobotTeleop/zenohapp_menu.jpeg:1]: I guess you are not going to use these images, I'd suggest we remove them from here. I also suggest that you add an image of the teleop fragment into the README, in order to let users know what they are going to find out when installing this app.

File: zenoh-android/RobotTeleop/app/src/main/res/layout/fragment_example.xml
Comment (DariusIMP) [inline zenoh-android/RobotTeleop/app/src/main/res/layout/fragment_example.xml:1]: It seems to me you are not using this, we could remove it.

File: zenoh-android/RobotTeleop/app/src/main/res/layout/fragment_gallery.xml
Comment (DariusIMP) [inline zenoh-android/RobotTeleop/app/src/main/res/layout/fragment_gallery.xml:1]: This seems to me can be removed, as well as the fragment_home.xml, fragment_slideshow.xml. I probably missed this on my original example.

**Review by DariusIMP — CHANGES_REQUESTED** [review]
I added a couple of comments. The app code could probably be improved but I understand our focus is to show how to use Zenoh in such a use case so I don't want to be too nit picky.

In general, besides the comments, I'd recommend to remove all the unnecessary commented code, which adds noise to the example, and to format the project [see how](https://www.jetbrains.com/help/idea/reformat-and-rearrange-code.html#reformat_module_directory).

**Pull request merged**